knowledge-base/manipulate-table-rows-based-on-their-position-in-the-physical-page.md

```diff
@@ -0,0 +1,156 @@
+
+For simplicity in the code we assume that all the measures are in inches. 
+
+```CSharp
+namespace ReportLibrary1
+{
+    using System;
+    using System.ComponentModel;
+    using System.Drawing;
+    using System.Linq;
+    using System.Reflection;
+    using Telerik.Reporting.Drawing;
+    using Telerik.Reporting.Processing;
+
+    /// <summary>
+    /// Summary description for Report1
+    /// </summary>
+    public partial class Report1 : Telerik.Reporting.Report
+    {
+        const int columnHeaderRowCount = 1;
+        readonly Color mainColor = Color.Red;
+        readonly Color secondaryColor = Color.DarkCyan;
+
+        readonly static PropertyInfo measurementsProp;
+        static PropertyDescriptor heightProp = null;
+
+        // Used to avoid multiple item color modifications
+        Telerik.Reporting.Processing.Report processingReportInstance = null;
+
+        static Report1()
+        {
+            measurementsProp = typeof(Telerik.Reporting.Processing.TextBox).GetProperties(BindingFlags.Instance | BindingFlags.NonPublic)
+                .First(pi => pi.Name == "Measurements");
+        }
+
+        public Report1()
+        {
+            //
+            // Required for telerik Reporting designer support
+            //
+            InitializeComponent();
+
+            //
+            // TODO: Add any constructor code after InitializeComponent call
+            //
+        }
+
+        private void pageHeaderSection1_ItemDataBinding(object sender, EventArgs e)
+        {
+            Telerik.Reporting.Processing.PageSection processingPageHeader = (Telerik.Reporting.Processing.PageSection)sender;
+            Telerik.Reporting.Processing.Report processingReport = processingPageHeader.Report;
+
+            if (this.processingReportInstance == processingReport)
+            {
+                return;
+            }
+            else
+            {
+                this.processingReportInstance = processingReport;
+            }
+
+            var useMainColor = true;
+
+            // Get the instance of the Processing Table
+            Telerik.Reporting.Processing.Table processingTable =
+                (Telerik.Reporting.Processing.Table)ElementTreeHelper.FindChildByName(processingReport, "table1", true)[0];
+
+            int tableColumnCount = this.table1.ColumnGroups.Count;
+            var tableColumnHeaderHeight = 0f;
+
+            int rowIndex = 0;
+            for (; rowIndex < columnHeaderRowCount; rowIndex++)
+            {
+                var row = processingTable.Rows[rowIndex];
+                var textBox = (Telerik.Reporting.Processing.TextBox)row.GetCell(0).Item;
+                tableColumnHeaderHeight = GetMeasuredHeight(textBox);
```

Comment:
the operator should be +=  or it will only account for the height of the last row of the column header.